be/src/runtime/file_result_writer.cpp

```diff
@@ -0,0 +1,319 @@
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "runtime/file_result_writer.h"
+
+#include "exec/broker_writer.h"
+#include "exec/local_file_writer.h"
+#include "exec/parquet_writer.h"
+#include "exprs/expr.h"
+#include "runtime/primitive_type.h"
+#include "runtime/row_batch.h"
+#include "runtime/tuple_row.h"
+#include "runtime/runtime_state.h"
+#include "util/types.h"
+#include "util/date_func.h"
+#include "util/uid_util.h"
+
+#include "gen_cpp/PaloInternalService_types.h"
+
+namespace doris {
+
+const size_t FileResultWriter::OUTSTREAM_BUFFER_SIZE_BYTES = 1024 * 1024;
+
+FileResultWriter::FileResultWriter(
+        const ResultFileOptions* file_opts,
+        const std::vector<ExprContext*>& output_expr_ctxs,
+        RuntimeProfile* parent_profile) :
+            _file_opts(file_opts),
+            _output_expr_ctxs(output_expr_ctxs),
+            _parent_profile(parent_profile) {
+}
+
+FileResultWriter::~FileResultWriter() {
+    close();
+}
+
+Status FileResultWriter::init(RuntimeState* state) {
+    _state = state;
+    _init_profile();
+
+    RETURN_IF_ERROR(_create_file_writer());
+    return Status::OK();
+}
+
+void FileResultWriter::_init_profile() {
+    RuntimeProfile* profile = _parent_profile->create_child("FileResultWriter", true, true);
+    _append_row_batch_timer = ADD_TIMER(profile, "AppendBatchTime");
+    _convert_tuple_timer = ADD_CHILD_TIMER(profile, "TupleConvertTime", "AppendBatchTime");
+    _file_write_timer = ADD_CHILD_TIMER(profile, "FileWriteTime", "AppendBatchTime");
+    _writer_close_timer = ADD_TIMER(profile, "FileWriterCloseTime");
+    _written_rows_counter = ADD_COUNTER(profile, "NumWrittenRows", TUnit::UNIT);
+    _written_data_bytes = ADD_COUNTER(profile, "WrittenDataBytes", TUnit::BYTES);
+}
+
+Status FileResultWriter::_create_file_writer() {
+    std::string file_name = _get_next_file_name();
+    if (_file_opts->is_local_file) {
+        _file_writer = new LocalFileWriter(file_name, 0 /* start offset */);
+    } else {
+        _file_writer = new BrokerWriter(_state->exec_env(),
+                _file_opts->broker_addresses,
+                _file_opts->broker_properties,
+                file_name,
+                0 /*start offset*/);
+    }
+    RETURN_IF_ERROR(_file_writer->open());
+
+    switch (_file_opts->file_format) {
+        case TFileFormatType::FORMAT_CSV_PLAIN:
+            // just use file writer is enough
+            break;
+        case TFileFormatType::FORMAT_PARQUET:
+            _parquet_writer = new ParquetWriterWrapper(_file_writer, _output_expr_ctxs);
```

Comment:
`_file_writer` seems owned by  `ParquetWriterWrapper`  by not realeased in `ParquetOutputStream` or `ParquetWriterWrapper` how about create and own it in `ParquetWriterWrapper`, and use use unique_prt to handle  `_file_writer`, `_parquet_writer` in this class

Reply:
`ParquetOutputStream` is not implemented.